libs/execution_context/ebpf_maps.c

```diff
@@ -606,6 +625,27 @@ _find_hash_map_entry(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key)
     return value;
 }
 
+static uint8_t*
+_find_and_delete_hash_map_entry(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key)
+{
+    uint8_t* value = NULL;
+    if (!map || !key)
+        return NULL;
+
+    if (ebpf_hash_table_find((ebpf_hash_table_t*)map->data, key, &value) != EBPF_SUCCESS) {
+        value = NULL;
+    }
+
+    if (ebpf_hash_table_delete((ebpf_hash_table_t*)map->data, key) != EBPF_SUCCESS) {
+        value = NULL;
+    }
```

Comment:
If I understand the code right, this is not atomic.   Find-and-delete needs to be atomic so the find and delete are both done under the same lock.

Reply:
Hash table delete is atomic. If the delete fails, we discard the found value. This makes the find and delete atomic.

Reply:
Got it, thanks